docs/gitbook/architecture/topology/patterns.adoc

```diff
@@ -0,0 +1,164 @@
+with one smart contract. It has no concern for privacy or throughput,
+which is suitable in the simplest cases -- for example, serving a static
+web app like a blog, or a grocery list blockchain shared by a family who
+all have equal access.
+
+
+== Molecules
+
+A _molecule_ is two or more atoms joined together for a particular
+purpose, forming various kinds of blockchain networks for privacy,
+scalability, security, and modularity.
+
+Beyond the basic molecules, your Interbit application can connect atoms
+to form molecules, atoms to molecules, and molecules to molecules, to
+form blockchain networks of arbitrary complexity.
+
+The molecules described in this section include:
+
+* <<_hub_and_spoke>>
+* <<_hub_and_connected_spoke>>
+* <<_control_by_proxy>>
+
+
+=== Hub and Spoke
+
+image:img/hub_and_spoke.svg["The Hub and Spoke molecule", 175, 194,
+role="right"]
+The Hub and Spoke molecule contains at least two atom types. It is used
+to increase throughput and/or to ensure privacy.
+
+In this molecule, the spoke atoms (in green, surrounding the hub) do not
+connect to each other nor do they share data directly; the spokes only
+connect with the hub (in blue). Each spoke can process actions
+dispatched from the hub, and the hub processes actions from every spoke.
+
+
+=== Hub and Connected Spoke
+
+image:img/hub_and_connected_spoke.svg["The Hub and Connected Spoke
+molecule", 175, 194, role="right"]
+The Hub and Connected spoke molecule has at least two atom types and is
+a more versatile version of the Hub and Spoke molecule. There is a
+greater potential for throughput increase, as some processing is
+offloaded from the hub onto the connected spokes. Privacy of spokes is
+assured, as sensitive information is passed directly to the concerned
+spokes.
+
+In this molecule, the hub could act as a directory blockchain providing
+public details about all of the connected spokes. When spokes want to
+communicate privately, they can create and connect to an intermediate
+blockchain that allows them to share data that neither the hub nor the
+other spokes could know about or access.
+
+
+=== Control by Proxy
+
+image:img/control_by_proxy.svg["The Control By Proxy molecule", 175,
+194, role="right"]
+The Control by Proxy molecule is a more-versatile version of a Hub and
+Spoke molecule that uses at least three atom types. One of the spokes is
+of a different type -- the _control_ atom (in red). The hub (in blue)
+acts as an intermediary. The spokes (in green) may not see any part of
+the control atom, and the control atom can have additional permissions
+in the hub that the spokes would not. As a result, the most sensitive
+data can be placed in the control atom without worry.
+
+This topology is useful, for example, for
+link:https://en.wikipedia.org/wiki/Health_Insurance_Portability_and_Accountability_Act[HIPAA]
+compliance. Imagine that the control atom is managed by a hospital.
+Patient information could flow to the spokes, each of which would be
```

Comment:
could flow to the spokes through the hub (aka. proxy) which only contains a subset of shareable information.

Reply:
This example contains the hidden implication that the control chain is itself a hub connecting a multitude of user chains and confuses the proxy concept a little bit. I think it is clearer to refer to the control chain as an administrator chain holding system metadata. 

Alternatively the example could be simplified. Instead of "patient data" it could say "A subset of the data" or some wording that does not imply another whole system running from the control chain.